pkg/app/piped/notifier/matcher.go

```diff
@@ -75,6 +93,15 @@ func (m *matcher) Match(event model.NotificationEvent) bool {
 			return false
 		}
 	}
+	// Should count current event as matched if it contains any of listening labels.
+	if len(m.labels) > 0 && len(labels) > 0 {
+		for k, v := range m.labels {
+			if ev, ok := labels[k]; ok && ev == v {
+				return true
+			}
+		}
+		return false
```

Comment:
I think it matches if and only if the event contains all the specified labels.  "all" instead of "any".
What do you think?

Reply:
I get your point. Tbh, I think one application/deployment may contain many labels for different purposes (just to filter on web or such), so making the filtering rule as "all" instead of "any" is a bit hard, especially in case the application contains many labels. That's why I make it this way, happy to hear other thoughts cc @knanao

Reply:
Addressed by https://github.com/pipe-cd/pipecd/pull/3241/commits/0a10e7be6a892af7f56f45fae53f6b440639357f